tket/src/Circuit/include/Circuit/ClassicalExpBox.hpp

```diff
@@ -23,13 +23,34 @@
 
 namespace tket {
 
+// Used as an intermediate class in between ClassicalExpBox and Box
+class ClassicalExpBoxPyBase : public Box {
+ public:
+  ClassicalExpBoxPyBase() : Box(OpType::ClassicalExpBox) {}
+  ClassicalExpBoxPyBase(const ClassicalExpBoxPyBase &other) : Box(other) {}
+  ~ClassicalExpBoxPyBase() override {}
+
+  /**
+   * @brief Rename the units in the logic expression according to the
+   * given Bit map. We can still mutate python objects even though the
+   * function is const.
+   *
+   * @param bm
+   * @return bool
+   */
+  virtual bool rename_units(const std::map<Bit, Bit> &bm) const {
+    (void)bm;
+    return false;
+  }
+};
+
 /**
  * Holding box for abstract expressions on Bits
  * Templated by type which holds expression
```

Comment:
As far as I can see the requirements for `T` (which we should document here) are the copy-constructor, the `equal()` and the `attr()` methods, but maybe it is simplest to just say that it must implement https://pybind11.readthedocs.io/en/stable/reference.html#_CPPv4I0E10object_api .